src/sensor/ping360.cpp

```diff
@@ -531,4 +526,23 @@ void Ping360::resetSettings()
     _sensorSettings.valid = false;
 }
 
+void Ping360::enableHeadingIntegration(bool enable)
+{
+    if (enable) {
+        connect(MavlinkManager::self(), &MavlinkManager::mavlinkMessage, this, &Ping360::processMavlinkMessage);
+    } else {
+        disconnect(MavlinkManager::self(), &MavlinkManager::mavlinkMessage, this, &Ping360::processMavlinkMessage);
+        _heading = 0;
+        emit headingChanged();
+    }
+}
+
+void Ping360::processMavlinkMessage(const mavlink_message_t& message)
+{
+    mavlink_attitude_t attitude;
+    mavlink_msg_attitude_decode(&message, &attitude);
```

Comment:
I think we should check the message type of `message` or change the type of the parameter to `const mavlink_attitude_t&`

Reply:
We only decode attitude messages, but done.